ROADMAP.md

```diff
@@ -0,0 +1,154 @@
+
+** Next Steps**
+ * Integrate serde based reflection
+ * Serialize/Deserialize ecs worlds
+
+### Streaming/Asset Handling
+TODO
+
+**Parallelism** TODO: Load assets in parallel
+**Data-Driven** TODO: low loading overhead
+
+** Next Steps**
+ * Improve current approach regarding robustness (error handling) and flexibility with scene formats
+ * Implement loaders for common file formats and archives
+
+### Documentation/Tutorial
+TODO
+
+** Next Steps**
+
+### Physics
+TODO: I would consider physics pretty difficult part to do *right*, therefore not part of the short term goals.
+Maybe rely on external libraries first.
+Should still provide ways for collision detection, allowing to do collision checks, raycasts, etc. (-> ncollide?).
+
+**Parallelism** TODO: internal physic parallelism, possible GPU implementation (hard), difficult with external libraries
+**Data-Driven** TODO: integration issue: physics world <-> ECS world, having 2 different representations? if so how to keep them in sync and integrate with other conflicting processors (e.g spatial hierarchy)
+
+** Next Steps**
+ * Integrate an existing library or write from scratch (requires further planning then)
+ * Provide collision testing API (triggers, raycasts, etc.)
+
+### Tooling
+We want to focus on developing a game engine, even though a rich toolset is required to simplify different task for users. If possible, we can integrate existing tools like profilers and 3D editors.
+'In-house' tools would be implemented on demand, e.g. engine specific data formats, own editor (Qt based?).
+
+** Next Steps**
+ * Integrate better logging (high priority -> short-term)
+
+### Scripting
+Possible scripting languages: mruby, lua, moonscript, dyon, ..
+
+Supporting a scripting language would support rapid-prototyping and faster iteration cycles. Also would be more attractive for developers who are not so much familiar with Rust.
+Scripting could be use for several subsystems like animation, particle system, AI, etc.
+
+** Next Steps**
+ * Evaluate the different scripting languages with respect to interoperability, performance, usability, ..
+ 
+### Graphical UI
+** Next Steps**
+ * Compatibility of GUIs with entity component systems (UI elements as entities <-> only root UI elements)
+ * Integrate an existing UI library (e.g conrod) or develop a new library from scratch (requires further planning then)
+ 
+Future:
+A common trend for large engines is embedding webbrowser frameworks like CEF for UI. Allowing to re-use existing JS/HTML/CSS frameworks to create complex UIs in a short time.
+Unfortunately CEF is pretty large, therefore a Servo based approach would be desirable for performance and easier integration.
+
+### Networking
+TODO: same as physics, hard to do correctly, interesting would be something like `libyojimbo` but well..
+
+** Next Steps**
+
+### AI
+TODO: also no real experience with this..
```

Comment:
AI support should be pretty minimal, I'm thinking something like A* implementations and the likes. How does this integrate with the physics engine?